### PR TITLE
Added parser for Spotlight store database (store.db) files #2093

### PR DESCRIPTION
**Pull request by joachimmetz**
Added parser for Spotlight store database (store.db) files #2093

### CODE REVIEW

File: plaso/parsers/spotlight_storedb.py [hunk 1]
```diff
@@ -0,0 +1,1050 @@
...
+      self._ReadPropertyPages(
+          file_object, file_header.metadata_types_block_number,
+          self._metadata_types)
+
+      self._metadata_values = {}
+      self._ReadPropertyPages(
+          file_object, file_header.metadata_values_block_number,
+          self._metadata_values)
+
+      self._ReadPropertyPages(
```
Comment (rgayon) [inline plaso/parsers/spotlight_storedb.py:1016]: Maybe add a comment that we're consuming this page but don't know what it is for

File: plaso/parsers/spotlight_storedb.py [hunk 2]
```diff
@@ -0,0 +1,1050 @@
...
+
+    file_offset += bytes_read
+
+    page_values_header = self._ReadStructureFromByteStream(
+        page_data, file_offset, data_type_map)
+
+    if page_header.property_table_type in (0x00000011, 0x00000021):
+      self._ReadPropertyPageValues(page_header, page_data, property_table)
+
+    elif page_header.property_table_type == 0x00000081:
```
Comment (rgayon) [inline plaso/parsers/spotlight_storedb.py:747]: What about 0x00000041 ?

File: plaso/parsers/spotlight_storedb.py [hunk 3]
```diff
@@ -0,0 +1,1050 @@
...
+    self.property_type = None
+    self.value = None
+    self.value_type = None
+
+
+class SpotlightStoreMetadataItem(object):
+  """Metadata item.
+
+  Attributes:
+    attributes (dict[str, MetadataAttribute]): metadata attributes.
```
Comment (rgayon) [inline plaso/parsers/spotlight_storedb.py:73]: SpotlightStoreMetadataAttribute

Comment (joachimmetz) [reply plaso/parsers/spotlight_storedb.py:73]: done

File: plaso/parsers/spotlight_storedb.py (hunk 2 above)
Comment (joachimmetz) [reply plaso/parsers/spotlight_storedb.py:747]: At the moment I don't know, I don't have a sample with 0x00000041 or more data about it.

File: plaso/parsers/spotlight_storedb.py (hunk 1 above)
Comment (joachimmetz) [reply plaso/parsers/spotlight_storedb.py:1016]: SG, done

**Comment by joachimmetz** [comment]
Replied to review comments and issues addressed @rgayon PTAL

**Pull request merged**